resources/openebs-operator-1.3.0.yaml

```diff
@@ -0,0 +1,751 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: openebs
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: openebs-maya-operator
+  namespace: openebs
+---
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: openebs
+  annotations:
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: docker/default
+    seccomp.security.alpha.kubernetes.io/defaultProfileName: docker/default
+spec:
+  privileged: true
+  seLinux:
+    rule: RunAsAny
+  supplementalGroups:
+    rule: RunAsAny
+  runAsUser:
+    rule: RunAsAny
+  fsGroup:
+    rule: RunAsAny
+  volumes:
+  - "*"
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: openebs-maya-operator
+rules:
+- apiGroups: ["extensions"]
+  resources: ["podsecuritypolicies"]
+  resourceNames: ["openebs"]
+  verbs: ["use"]
+- apiGroups: ["*"]
+  resources: ["nodes", "nodes/proxy"]
+  verbs: ["*"]
+- apiGroups: ["*"]
+  resources: ["namespaces", "services", "pods", "pods/exec", "deployments", "replicationcontrollers", "replicasets", "events", "endpoints", "configmaps", "secrets", "jobs", "cronjobs"]
```

Comment:
This seems like some pretty wide permissions? Does the operator really need the ability to exec into every pod, or interact read/write/update on pretty much all the core objects in the entire cluster?

Reply:
This is the spec from their install instructions (https://openebs.github.io/charts/openebs-operator-1.3.0.yaml), I can't speak to whether all of these are really used or not.

Reply:
Well, it seems overly broad, so I sort of think we shouldn't include unless we've reasonably hardening the configuration. Otherwise we shouldn't really market gravity as a hardened distribution as we're currently doing.

Reply:
Yep, that's fair. I'll see if I can find out whether all of these are really needed or not.

Reply:
@knisbet I've tweaked RBAC:

* All overly broad permissions (such as for core objects) are contained to the `openebs` namespace.
* Access to OpenEBS-specific custom resources + storage-related ones is allowed cluster-wide.
* Removed some permissions that didn't seem to make sense to me (like, autoscalers, ingresses, etc. - not sure why OpenEBS would need them).

Things seem to still work fine with this configuration. We can tweak it as we go if we run into any issues.